src/emulator/storage/files.ts

```diff
@@ -125,12 +156,72 @@ export class StorageLayer {
     private _cloudFunctions: StorageCloudFunctions
   ) {}
 
+  /**
+   * Create SignedUrl
+   * @date 6/28/2023 - 2:56:09 PM
+   *
+   * @async
+   * @param {GetObjectRequest} request
+   * @returns {Promise<boolean>}
+   */
+  async generateSignedUrl(request: CreateSignedUrl): Promise<SignedUrlResponse> {
+    const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
+
+    const currentDate = this.getCurrentDate();
+
+    const timeToLive = request.ttlSeconds;
+    if (!Number.isInteger(timeToLive)) {
+      throw new BadRequestError("Invalid TTL Parameter");
```

Comment:
nit: Have a more descriptive error: "Invalid TTL: ${timeToLive} is not an integer."